react/features/chat/components/native/Chat.js

```diff
@@ -0,0 +1,196 @@
+ *
+ * Co-authored with https://github.com/dgeorgiev06
+ */
+type Props = {
+
+    /**
+     * True if the chat window should be open.
+     */
+    _isOpen: boolean,
+
+    /**
+     * The local participant.
+     */
+    _localParticipant: Object,
+
+    /**
+     * All the chat messages in the conference.
+     */
+    _messages: Array<Object>,
+
+    /**
+     * Toggles the chat window.
+     */
+    _onToggleChat: Function,
+
+    /**
+     * Send a text message.
+     *
+     * @protected
+     */
+    _onSendMessage: Function
+};
+
+
+/**
+ * The chat page of the mobile (i.e. React Native) application.
+ */
+class Chat extends Component<Props> {
+
+    /**
+       * Initializes a new Conference instance.
+       *
+       * @param {Object} props - The read-only properties with which the new
+       * instance is to be initialized.
+       */
+    constructor(props: Props) {
+        super(props);
+        this._onSend = this._onSend.bind(this);
+        this._renderMessage = this._renderMessage.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *x
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+
+        const messages = this.props._messages.map(this._renderMessage);
+
+        return (
+            <Modal
+                animationType = { 'slide' }
+                onRequestClose = { this.props._onToggleChat }
+                supportedOrientations = { [
+                    'landscape',
+                    'portrait'
+                ] }
+                transparent = { true }
+                visible = { this.props._isOpen }>
+                <Header>
+                    <BackButton onPress = { this.props._onToggleChat } />
+                    <HeaderLabel labelKey = 'chat.title' />
+                </Header>
+                <SafeAreaView style = { styles.modalBackdrop }>
+                    <GiftedChat
+                        messages = { messages.reverse() }
+                        onSend = { this._onSend }
+                        user = {{ _id: this.props._localParticipant.id }} />
+                </SafeAreaView>
+            </Modal>
+        );
+    }
+
+    _onSend: (Array<Object>) => void;
+
+    /**
+     * Called by {@code render} to create the chat div.
+     *
+     * @param {string} message - The chat message to display.
+     * @param {string} id - The chat message ID to use as a unique key.
+     * @returns {Array<ReactElement>}
+     */
+    _onSend([ message ]) {
+        this.props._onSendMessage(message.text);
+    }
+
+    _renderMessage: (Object) => void;
+
+    /**
+     * Called by {@code _onSubmitMessage} to create the chat message.
+     *
+     * @param {string} message - The chat message to display.
+     * @param {string} id - The chat message ID to use as a unique key.
+     * @returns {Array<ReactElement>}
+     */
+    _renderMessage(message: Object, id: string) {
```

Comment:
Avatars are not our avatars. We can use our avatars by putting the URL in the user field.

Reply:
I'm reworking the layout, please review after the next push and let's see how much (more?) does that fit to Jitsi